app/services/authentication/providers/github.rb

```diff
@@ -0,0 +1,34 @@
+module Authentication
+  module Providers
+    # GitHub authentication provider, uses omniauth-github as backend
+    class Github < Provider
```

Comment:
extracted all the provider specific logic from the previous `AuthorizationService` to a provider class